src/library.coffee

```diff
@@ -122,21 +137,28 @@ class Library extends EventEmitter
     return null
 
   _updateComponents: (components) ->
-    names = Object.keys components
+    names = []
     for name, comp of components
       if not comp
         # removed
         @components[name] = null
+        names.push name if names.indexOf(name) is -1
         continue
+      discovered = cleanComponentDefinition comp
       existing = @getComponent name
-      unless existing
+      unless existing?.definition
         # added
-        @components[name] = comp
+        @components[name] = discovered
+        names.push name if names.indexOf(name) is -1
+        continue
+      unless JSON.stringify(existing.definition) is JSON.stringify(discovered.definition)
+        # updated
+        @components[name] = discovered
+        names.push name if names.indexOf(name) is -1
         continue
-      # update
-      for k, v of comp
```

Comment:
Doesn't this destroy `command` ? which is needed for instantiating component instances?